README.md
[![DOI](https://zenodo.org/badge/166169099.svg)](https://zenodo.org/badge/latestdoi/166169099)

# Chinese Ideophone Database

<img src='https://github.com/simazhi/chinese_ideophone_database/blob/master/badge/logo.png' align="right" height="139" />

## Current version

**CURRENT VERSION OF THE DATABASE: 1.0**

## Link to the paper that details the contents


[Van Hoey, Thomas & Arthur Lewis Thompson. 2020. The Chinese Ideophone Database (CHIDEOD). *Cahiers de linguistique Asie orientale* 49(2). 136–167.](https://brill.com/view/journals/clao/49/2/article-p136_2.xml) ([doi](https://doi.org/10.1163/19606028-bja10006))

## Deployments of the Chinese Ideophone Database

This is the directory where the R data package CHIDEOD is stored.
**PLEASE NOTE THAT THE MAIN PROJECT REPOSITORY IS HELD AT THE OSF REPOSITORY (https://osf.io/kpwgf/) AND NOT HERE,** although you are free to look around at the scripts that are used to update the Chinese Ideophone Database.

If you navigate to the OSF-repository, you can access the Chinese Ideophone Database (CHIDEOD) in the following formats:

* **chideod.rds**
* **chideod.csv**
* **chideod.xlsx**

As a further note, you may found similar files here in this repository. It is not recommended to use these, but always go to the OSF repository.

Other ways of accessing CHIDEOD is through the 

* **[app deployment](https://simazhi.shinyapps.io/chideod_appversion/)** (look under the Wiki tab for a guaranteed link in case this one breaks down).
* as an R package.

The R package can be downloaded as follows:

```
library(devtools)
devtools::install_github("simazhi/chinese_ideophone_database/CHIDEOD")
```

## Contents of this github folder

### Files

* __README.md__ The file you are reading right now.
* __LICENCE.md__ The file that contains the license of this project.
* __chinese_ideophone_database.Rproj__ This file is used when we update CHIDEOD, and allows for easy project management. You can ignore this file.
* __CHIDEOD_VERSIONNUMBER_tar.gz__ This file contains the raw R package of CHIDEOD. You can use this if the above instructions to install the R package fail.
* __.gitignore__ This file tells github which files to ignore. You can do so too.
* __Changelog.md__ This file contains the changes made after CHIDEOD 1.0, which is congruent with the version as published in the explanatory paper (*add link to article*).

### Folders

* __CHIDEOD__ This folder contains the contents of the R package for CHIDEOD, before it is zipped in the tar.gz file. It is not recommended to go inside this.
* __interesting_lists__ This folder contains lists that were deemed interesting for us when making the Chinese Ideophone Database, and holds miscellaneous items and statistics that were made using previous stages of the database. Got a free afternoon and want to go through it for ideophone treasurehunting? Sure, go ahead.
* __scripts__ A misscelaneous collection of scripts from which we draw when updating the database. Feel free to reuse parts of them if they can help you solve your own database problems. On the other hand, please do not expect us to list the contents of each one. They are mostly intented for our own ad hoc use and adaptation and we provide them here for our own open transparency.
* __shiny_app__ The code for the shiny app deployment of CHIDEOD.
* __variables_and_data__ This folder contains the basic structure of CHIDEOD (chideod_variables.pdf), as well as a list of abbreviations and references (abbreviations_references.md), used in the database itself. **We strongly recommend to read the paper version, which goes in detail over all of these items.**

# License

This product is intended as open-source and non-commercial.
The license is the [CC BY-NC-SA 4.0 license](https://creativecommons.org/licenses/by-nc-sa/4.0/). 
Please consult the LICENCE.md file for more information.

# How to cite

Please cite the reference paper

**Van Hoey, Thomas & Arthur Lewis Thompson. 2020. The Chinese Ideophone Database (CHIDEOD). Cahiers de linguistique Asie orientale 49(2). 136–167. https://doi.org/10.1163/19606028-bja10006.**

And/or the current project version, based on the OSF repository:

**Van Hoey, Thomas & Arthur Lewis Thompson. 2019. Chinese ideophone database (CHIDEOD). doi: 10.17605/OSF.IO/KPWGF**

If used in a project, you can also add the version number (see top of this README.md file).



Changelog.md
# Changelog

This file serves to record future updates, after the article which contains the baseline database (v.1.0) is published. ([link to article](https://brill.com/view/journals/clao/49/2/article-p136_2.xml))

# 1.0

This version is essentially the same data as 0.9.3, but counts as the baseline for future updates.

zdic/zdicgetter.R
#library(tidyverse)
library(dplyr)
library(tibble)
library(magrittr)
library(readr)
library(purrr)
library(rvest)
library(glue)
library(here)

# function definition

zdicfinder <- function(SEARCHWORD){
  
  url <- glue("https://www.zdic.net/hans/{SEARCHWORD}")

  cat(glue("{SEARCHWORD}      "))
  
  page <- read_html(url)
  finaltable <- page %>%
    html_nodes('[class="gycd-item"]') %>%
    #html_nodes(xpath = '//*[@id="126006_1"]/div[2]/div[3]/div/ol/div[1]') %>%
    html_text() %>%
    enframe() %>%
    mutate(ideo = SEARCHWORD)

  write_csv(finaltable, here(#"zdic", 
			     "zdicwords.csv"),
              append = TRUE)
  
  Sys.sleep(5) # sleep 5 seconds
}

# wrap in possibly
zdicfinder_poss <- possibly(zdicfinder, otherwise = NA)

# read in the list of ideophones
targets <- read_csv(here(#"zdic",
                         "zdicideophonestargets.csv")) %>% pull() %>% sort()
targets2 <- c("咻咻", "汪汪")

# run function
walk(targets2, zdicfinder_poss)
walk(targets, zdicfinder_poss)



CHIDEOD/R/CHIDEOD.R
#' CHIDEOD
#' A data package for CHIDEOD.
#' @docType package
#' @aliases CHIDEOD-package
#' @title CHIDEOD -- The Chinese Ideophone Database
#' @name CHIDEOD
#' @description This package contains the Chinese Ideophone Database CHIDEOD.
#' @details Use \code{data(package='CHIDEOD')$results[, 3]} to see a list of available data sets in this data package
#'     and/or DataPackageR::load_all
#' _datasets() to load them.
#' @seealso
#' \link{chideod}
NULL




#' Detailed description of the data
#' @name chideod
#' @docType data
#' @title Descriptive data title
#' @format a \code{data.frame} containing the following fields:
#' \describe{
#' \item{traditional}{}
#' \item{simplified}{}
#' \item{pinyin_tone}{}
#' \item{pinyin_tonenumber}{}
#' \item{pinyin_without_tone}{}
#' \item{ipa_toneletter}{}
#' \item{ipa_tonenumber}{}
#' \item{middle_chinese_baxter}{}
#' \item{middle_chinese_ipa}{}
#' \item{old_chinese_ipa}{}
#' \item{language_stage}{}
#' \item{data_source}{}
#' \item{traditional1}{}
#' \item{traditional2}{}
#' \item{traditional3}{}
#' \item{traditional4}{}
#' \item{simplified1}{}
#' \item{simplified2}{}
#' \item{simplified3}{}
#' \item{simplified4}{}
#' \item{sensory_imagery}{}
#' \item{definitions}{}
#' \item{orthograpic_variants}{}
#' \item{morphological_template}{}
#' \item{radical_support}{}
#' \item{character1_freq}{}
#' \item{character2_freq}{}
#' \item{character3_freq}{}
#' \item{character4_freq}{}
#' \item{character1_family_size}{}
#' \item{character2_family_size}{}
#' \item{character3_family_size}{}
#' \item{character4_family_size}{}
#' \item{character1_semantic_radical}{}
#' \item{character2_semantic_radical}{}
#' \item{character3_semantic_radical}{}
#' \item{character4_semantic_radical}{}
#' \item{character1_semantic_radical_freq}{}
#' \item{character2_semantic_radical_freq}{}
#' \item{character3_semantic_radical_freq}{}
#' \item{character4_semantic_radical_freq}{}
#' \item{character1_semantic_family_size}{}
#' \item{character2_semantic_family_size}{}
#' \item{character3_semantic_family_size}{}
#' \item{character4_semantic_family_size}{}
#' \item{character1_phonetic_component}{}
#' \item{character2_phonetic_component}{}
#' \item{character3_phonetic_component}{}
#' \item{character4_phonetic_component}{}
#' \item{character1_phonetic_component_freq}{}
#' \item{character2_phonetic_component_freq}{}
#' \item{character3_phonetic_component_freq}{}
#' \item{character4_phonetic_component_freq}{}
#' \item{character1_phonetic_family_size}{}
#' \item{character2_phonetic_family_size}{}
#' \item{character3_phonetic_family_size}{}
#' \item{character4_phonetic_family_size}{}
#' \item{interjection}{}
#' \item{note}{}
#' }
#' @source The data comes from https://osf.io/kpwg.
#' @seealso
#' \link{CHIDEOD}
NULL


CHIDEOR/R/CHIDEOR.R
#' CHIDEOR
#' A data package for CHIDEOD.
#' @docType package
#' @aliases CHIDEOR-package
#' @title CHIDEOD -- The Chinese Ideophone Database
#' @name CHIDEOR
#' @description A description of the data package
#' @details Use \code{data(package='CHIDEOR')$results[, 3]} to see a list of available data sets in this data package
#'     and/or DataPackageR::load_all
#' _datasets() to load them.
#' @seealso
#' \link{chideod}
NULL




#' Detailed description of the data
#' @name chideod
#' @docType data
#' @title The Chinese Ideophone Database (CHIDEOD)
#' @format a \code{tbl_df} containing the following fields:
#' \describe{
#' \item{traditional}{}
#' \item{simplified}{}
#' \item{pinyintone}{}
#' \item{pinyinnum}{}
#' \item{pinyinnone}{}
#' \item{HYDCD}{}
#' \item{zdic}{}
#' \item{Kroll}{}
#' \item{MC}{}
#' \item{OC}{}
#' \item{radsup}{}
#' \item{S1}{}
#' \item{S2}{}
#' \item{S3}{}
#' \item{S4}{}
#' \item{T1}{}
#' \item{T2}{}
#' \item{T3}{}
#' \item{T4}{}
#' \item{morphology}{}
#' \item{S1.charfreq}{}
#' \item{S2.charfreq}{}
#' \item{S3.charfreq}{}
#' \item{S4.charfreq}{}
#' \item{S1.famfreq}{}
#' \item{S2.famfreq}{}
#' \item{S3.famfreq}{}
#' \item{S4.famfreq}{}
#' \item{S1.sem}{}
#' \item{S2.sem}{}
#' \item{S3.sem}{}
#' \item{S4.sem}{}
#' \item{S1.semfreq}{}
#' \item{S2.semfreq}{}
#' \item{S3.semfreq}{}
#' \item{S4.semfreq}{}
#' \item{S1.semfam}{}
#' \item{S2.semfam}{}
#' \item{S3.semfam}{}
#' \item{S4.semfam}{}
#' \item{S1.phon}{}
#' \item{S2.phon}{}
#' \item{S3.phon}{}
#' \item{S4.phon}{}
#' \item{S1.phonfreq}{}
#' \item{S2.phonfreq}{}
#' \item{S3.phonfreq}{}
#' \item{S4.phonfreq}{}
#' \item{S1.phonfam}{}
#' \item{S2.phonfam}{}
#' \item{S3.phonfam}{}
#' \item{S4.phonfam}{}
#' \item{sensorymod}{}
#' \item{variant}{}
#' \item{datasource}{}
#' \item{note}{}
#' }
#' @source The data comes from https://osf.io/kpwg.
#' @seealso
#' \link{CHIDEOR}
NULL


scripts/frequencies.ASBC.R
start <- Sys.time()
library(tidyverse)
library(fs)
library(glue)
# library(furrr)
# future::plan(multisession)

words <- read_csv("list.of.traditional.ideophones.csv") 
words <- words %>%
  mutate(traditional = case_when(
    nchar(traditional) < 2 ~ str_replace(traditional, "(.)", " \\1_"),
    TRUE ~ traditional
  )) 
search.expressions <- pull(words)
#search.expressions

# files <- dir_ls("/Users/Thomas/Desktop/ASBC/tidy_ASBC_unnested",
#                     recurse = TRUE,
#                     regexp = ".rds$")

files <- dir_ls("tidy_ASBC_unnested",
                recurse = TRUE,
                regexp = ".rds$")

#search.expressions <- c("十二月", "必須")
search.2 <- glue_collapse(search.expressions, sep = "|")
search.2

# argList <- list(x = search.2, y = head(files))
# crossArg <- cross_df(argList)


finder <- function(file) {
  
  # read in file, select the variables I want
  current.file <- read_rds(file) %>% select(class, article.no, text)
  
  # is the search expression in the file
  final.file <- filter(current.file, str_detect(text, search.2))
  
  # write final table to the output file
  write_csv(final.file, "finder.results.csv", append = TRUE)
  #final.file

}


#map2_df(crossArg$x, crossArg$y, finder)
walk(files, finder)
#map2_dfr(crossArg$y, crossArg$x, finder)
end <- Sys.time()
end - start

# this is for reduced dataset
# add new column with search expression
#final.table <- mutate(filter.file, searchword = search.expression)


## POST PROCESSING
# map2_dfr(crossArg$y, crossArg$x, finder) -> r
# 
# r %>%
#   mutate(t = str_extract_all(text,search.2)) %>%
#   #select(t) %>%
#   unnest(t)





CHIDEOD/data-raw/documentation.R
#' CHIDEOD
#' A data package for CHIDEOD.
#' @docType package
#' @aliases CHIDEOD-package
#' @title CHIDEOD -- The Chinese Ideophone Database
#' @name CHIDEOD
#' @description This package contains the Chinese Ideophone Database CHIDEOD.
#' @details Use \code{data(package='CHIDEOD')$results[, 3]} to see a list of available data sets in this data package
#'     and/or DataPackageR::load_all
#' _datasets() to load them.
#' @seealso
#' \link{chideod}
NULL



#' Detailed description of the data
#' @name chideod
#' @docType data
#' @title Descriptive data title
#' @format a \code{data.frame} containing the following fields:
#' \describe{
#' \item{traditional}{}
#' \item{simplified}{}
#' \item{pinyin_tone}{}
#' \item{pinyin_tonenumber}{}
#' \item{pinyin_without_tone}{}
#' \item{ipa_toneletter}{}
#' \item{ipa_tonenumber}{}
#' \item{middle_chinese_baxter}{}
#' \item{middle_chinese_ipa}{}
#' \item{old_chinese_ipa}{}
#' \item{language_stage}{}
#' \item{data_source}{}
#' \item{traditional1}{}
#' \item{traditional2}{}
#' \item{traditional3}{}
#' \item{traditional4}{}
#' \item{simplified1}{}
#' \item{simplified2}{}
#' \item{simplified3}{}
#' \item{simplified4}{}
#' \item{sensory_imagery}{}
#' \item{definitions}{}
#' \item{orthograpic_variants}{}
#' \item{morphological_template}{}
#' \item{radical_support}{}
#' \item{character1_freq}{}
#' \item{character2_freq}{}
#' \item{character3_freq}{}
#' \item{character4_freq}{}
#' \item{character1_family_size}{}
#' \item{character2_family_size}{}
#' \item{character3_family_size}{}
#' \item{character4_family_size}{}
#' \item{character1_semantic_radical}{}
#' \item{character2_semantic_radical}{}
#' \item{character3_semantic_radical}{}
#' \item{character4_semantic_radical}{}
#' \item{character1_semantic_radical_freq}{}
#' \item{character2_semantic_radical_freq}{}
#' \item{character3_semantic_radical_freq}{}
#' \item{character4_semantic_radical_freq}{}
#' \item{character1_semantic_family_size}{}
#' \item{character2_semantic_family_size}{}
#' \item{character3_semantic_family_size}{}
#' \item{character4_semantic_family_size}{}
#' \item{character1_phonetic_component}{}
#' \item{character2_phonetic_component}{}
#' \item{character3_phonetic_component}{}
#' \item{character4_phonetic_component}{}
#' \item{character1_phonetic_component_freq}{}
#' \item{character2_phonetic_component_freq}{}
#' \item{character3_phonetic_component_freq}{}
#' \item{character4_phonetic_component_freq}{}
#' \item{character1_phonetic_family_size}{}
#' \item{character2_phonetic_family_size}{}
#' \item{character3_phonetic_family_size}{}
#' \item{character4_phonetic_family_size}{}
#' \item{interjection}{}
#' \item{note}{}
#' }
#' @source The data comes from https://osf.io/kpwg.
#' @seealso
#' \link{CHIDEOD}
NULL



shiny_app/app.R
#
# This is a Shiny web application. You can run the application by clicking
# the 'Run App' button above.
#
# Find out more about building applications with Shiny here:
#
#    http://shiny.rstudio.com/
#

#global.R
library(shiny)
library(tidyverse)
#library(readxl)
library(DT)
library(shinythemes)

# Here is the data and stuff

#library(devtools)
#install_github("simazhi/chinese_ideophone_database/CHIDEOR")

library(CHIDEOD) 
#make sure this is the latest version of this package
data("chideod")

#ideodata <- read_xlsx("www/data.xlsx") %>%
ideodata <- chideod



## SERVER

server <- 
    function(input, output) {
        checkGroup <- reactive(c(input$checkGroupDatavars,
                                 input$checkGroupOrthography,
                                 input$checkGroupPhonology,
                                 input$checkGroupMeaning,
                                 input$checkGroupMotivation,
                                 input$checkGroupFormation,
                                 input$checkGroupFrequencies,
                                 input$checkGroupOther))
        #input$checkGroup <- c(input$checkGroupForm,input$checkGroupMeaning)

        # Filter data based on selections
        output$table <- DT::renderDataTable(
            DT::datatable({
                data <- ideodata
                data <- data[, checkGroup(), drop = FALSE]
                data <- distinct(data)
            },
            extensions = c('Buttons', 'ColReorder'), 
            options = list(dom = 'Bfrtip',
                           buttons = c('copy', 
                                       'csv', 
                                       'excel'#, 
                                       # 'pdf', 
                                       # 'print'
                                       ),
                           colReorder = TRUE)
            )
        )


    
}



## UI

ui <-
navbarPage("CHIDEOD",
           fluid = TRUE,
           theme = shinytheme("flatly"),
           tabPanel("Table",
                    sidebarLayout(position = "left",
                                  #fluid = TRUE,
                                  sidebarPanel(
                                      width = 3,
                                      #checkboxes
                                      checkboxGroupInput("checkGroupDatavars", 
                                                         label = h2("Data variables"), 
                                                         choices = c("Language stage" = "language_stage",
                                                                     "Data source" = "data_source"),
                                                         selected = c("language_stage", "data_source")
                                      ),
                                      #end of checkboxes
                                      br(),
                                      h2("Descriptive variables"),
                                      #checkboxes
                                      checkboxGroupInput("checkGroupPhonology", 
                                                         label = h3("Phonology"), 
                                                         choices = c("Pinyin without tones" = "pinyin_without_tone", 
                                                                     "Pinyin with tones" = "pinyin_tone",
                                                                     "Pinyin with numbers" = "pinyin_tonenumber",
                                                                     "IPA with toneletters"= "ipa_toneletter",
                                                                     "IPA with tonenumbers" = "ipa_tonenumber",
                                                                     "Middle Chinese Baxter" = "middle_chinese_baxter", 
                                                                     "Middle Chinese IPA" = "middle_chinese_ipa",
                                                                     "Old Chinese IPA" = "old_chinese_ipa"),
                                                         selected = "pinyin_without_tone"
                                      ),
                                      #end of checkboxes
                                      br(),                                      
                                      #checkboxes two
                                      checkboxGroupInput("checkGroupOrthography", 
                                                         label = h3("Orthography"), 
                                                         choices = c("Traditional Chinese" = "traditional",
                                                                     "Simplified Chinese" = "simplified",
                                                                     "Traditional character 1" = "traditional1",
                                                                     "Traditional character 2" = "traditional2",
                                                                     "Traditional character 3" = "traditional3",
                                                                     "Traditional character 4" = "traditional4",
                                                                     "Simplified character 1" = "simplified1",
                                                                     "Simplified character 2" = "simplified2",
                                                                     "Simplified character 3" = "simplified3",
                                                                     "Simplified character 4" = "simplified4",
                                                                     "Semantic radical 1" = "character1_semantic_radical",
                                                                     "Semantic radical 2" = "character2_semantic_radical",
                                                                     "Semantic radical 3" = "character3_semantic_radical",
                                                                     "Semantic radical 4" = "character4_semantic_radical",
                                                                     "Sound component 1" = "character1_phonetic_component",
                                                                     "Sound component 2" = "character2_phonetic_component",
                                                                     "Sound component 3" = "character3_phonetic_component",
                                                                     "Sound component 4" = "character4_phonetic_component",
                                                                     "Orthographic variants" = "orthograpic_variants"),                                                   
                                                         selected = "traditional"
                                      ),
                                      #end of checkboxes
                                      br(),                                    
                                      # checkboxes five
                                      checkboxGroupInput("checkGroupMeaning", 
                                                         label = h3("Definitions"), 
                                                         choices = c("Definitions" = "definitions")
                                      ),
                                      #end of checkboxes
                                      br(),
                                      # checkboxes three
                                      checkboxGroupInput("checkGroupFormation", 
                                                         label = h2("Analytical variables"), 
                                                         choices = c("Morphological template" = "morphological_template",
                                                                     "Radical support" = "radical_support",
                                                                     "Interjection" = "interjection",
                                                                     "Sensory imagery" = "sensory_imagery")
                                      ),
                                      #end of checkboxes
                                      br(),
                                      # checkboxes six
                                      checkboxGroupInput("checkGroupFrequencies", 
                                                         label = h3("Frequency measures"), 
                                                         choices = c("Character 1 freq" = "character1_freq",
                                                                     "Character 2 freq" = "character2_freq",
                                                                     "Character 3 freq" = "character3_freq",
                                                                     "Character 4 freq" = "character4_freq",
                                                                     "Character 1 family size" = "character1_family_size",
                                                                     "Character 2 family size" = "character2_family_size",
                                                                     "Character 3 family size" = "character3_family_size",
                                                                     "Character 4 family size" = "character4_family_size",
                                                                     "Character 1 radical freq" = "character1_semantic_radical_freq",
                                                                     "Character 2 radical freq" = "character2_semantic_radical_freq",
                                                                     "Character 3 radical freq" = "character3_semantic_radical_freq",
                                                                     "Character 4 radical freq" = "character4_semantic_radical_freq",
                                                                     "Character 1 radical family size" = "character1_semantic_family_size",
                                                                     "Character 2 radical family size" = "character2_semantic_family_size",
                                                                     "Character 3 radical family size" = "character3_semantic_family_size",
                                                                     "Character 4 radical family size" = "character4_semantic_family_size",
                                                                     "Character 1 phonetic freq" = "character1_phonetic_component_freq",
                                                                     "Character 2 phonetic freq" = "character2_phonetic_component_freq",
                                                                     "Character 3 phonetic freq" = "character3_phonetic_component_freq",
                                                                     "Character 4 phonetic freq" = "character4_phonetic_component_freq",
                                                                     "Character 1 phonetic family size" = "character1_phonetic_family_size",
                                                                     "Character 2 phonetic family size" = "character2_phonetic_family_size",
                                                                     "Character 3 phonetic family size" = "character3_phonetic_family_size",
                                                                     "Character 4 phonetic family size" = "character4_phonetic_family_size")
                                      ),
                                      #end of checkboxes
                                      br(),
                                      # checkboxes seven
                                      checkboxGroupInput("checkGroupOther", 
                                                         label = h3("Other variables"), 
                                                         choices = c("Notes" = "note")
                                      )
                                      #, #end of checkboxes
                                      # downloadButton('download',"Download the data")
                                  ), # end of sidepanel
                                  
                                  mainPanel(width = 9,
                                            span(style='color:blue',
                                            h3("How to use this app"),
                                            print("Select the variables you are interested in by checking the boxes. You can then export the results or further trim them by inputting a search query. Note that if you search for phonological variables, you need to add a tilde to separate the syllables, e.g., 'ling~ling'.")
                                            ),
                                            br(),
                                            br(),
                                            DT::dataTableOutput("table")
                                  ) #main panel
                    ), #sidebar layout,
                    hr(),
                    print("© Thomas Van Hoey 司馬智"),
                    a(href="https://www.thomasvanhoey.com", "(website)"),
                    print("and Arthur Lewis Thompson"),
                    br(),
                    print("Last updated:"),
                    strong("March 2020, CHIDEOD (0.9.3)"),
                    a(href="https://osf.io/kpwgf/", "(link to repository)"),
                    br(),
                    print("Please cite as: "),
                    strong("Van Hoey, Thomas & Arthur Lewis Thompson. 2019. Chinese ideophone database. osf.io/kpwgf")
                    
           ),
           tabPanel("About",
                    h2("About CHIDEOD"),
                    h4("— The Chinese Ideophone Database"),
                    p("The Chinese Ideophone Database (CHIDEOD) is an open-source dataset, which currently collects 4948 unique onomatopoeia and ideophones (mimetics) of Mandarin, as well as Middle Chinese and Old Chinese. These are analyzed according to a wide range of variables pertaining to the type of data, description, analysis and frequency."),
                    br(),
                    print("© Thomas Van Hoey 司馬智"),
                    a(href="https://www.thomasvanhoey.com", "(website)"),
                    print("and Arthur Lewis Thompson"),
                    br(),
                    br(),
                    print("Last updated:"),
                    strong("June 2020"),
                    a(href="https://osf.io/kpwgf/", "(link to repository)"),
                    br(),
                    br(),
                    print("Link to the explanatory paper (to be added)."),
                    br(),
                    br(),
                    h2("Please cite as: "),
                    strong("Van Hoey, Thomas & Arthur Lewis Thompson. 2019. Chinese ideophone database (CHIDEOD). doi: 10.17605/OSF.IO/KPWGF"),
                    br(),
                    h2("Licence"),
                    print("The database is licenced under "),
                    a(href="https://creativecommons.org/licenses/by-nc-sa/4.0/", "the CC BY-NC-SA 4.0 licence"),
                    print(". This means that you can share and adapt, but that you cannot use it for any commerical purposes, and that you need to attribute correctly."),
                    br(),
                    h2("Disclaimer"),
                    print("CHIDEOD has been devised to serve academic research, especially within the domain of ideophone studies. We recommend CHIDEOD as an exploratory tool whilst consulting the datasets on the CHIDEOD Open Science Framework page. All CHIDEOD data were collected with fair use in mind and suggest that users pay attention to the ‘Data Source’ listed in CHIDEOD so that they may also cite and examine the primary sources behind CHIDEOD. We also ask that users and visitors cite CHIDEOD if it is has aided, benefited, or contributed to their research. If any typos or mistakes are encountered while using CHIDEOD, please do not hesitate to contact us so that we can provide a better repository for the study of Chinese ideophones.")

                    ),
           tabPanel("Data variables and abbreviations",
                    includeMarkdown("https://raw.githubusercontent.com/simazhi/chinese_ideophone_database/master/variables_and_data/abbreviations_references.md"))
           )


# fluidPage(theme = shinytheme("flatly"),
#     titlePanel("The Chinese Ideophones Database (CHIDEOD)"),
    

    
#) #end tabpage




# Run the application 
shinyApp(ui = ui, server = server)
